tct/src/internal/hash/fast_hash.rs

```diff
@@ -0,0 +1,25 @@
+//! This module, when enabled with the `fast_hash` feature flag, replaces the Poseidon hash
+//! implementation with `blake2b_simd`, which is much faster. **This is useless for production use
+//! of this crate; it is only useful to accelerate testing, when we don't care about producing
+//! zero-knowledge proofs.**
```

Comment:
We shouldn't make this change in this PR, because it is already far too large, but wouldn't this be a non-problem if the tree were generic over the leaf type, like the `incrementalmerkletree` API?